### ISSUE TITLE
Paper Advance to Half Position

### ISSUE THREAD

**Issue by Fourniture** [musescore/MuseScore#25836]
### Your idea

When entering the last notes on a page, the paper will immediately jump to the next page. However, this is often unhelpful if someone is trying to write the final note for _all_ staves before moving to the next page.

https://github.com/user-attachments/assets/3837c7b5-53ae-4549-a3f6-117dbdf9a9d1

I suggest that the paper, instead of advancing to the next page, advances to either the half position (as I dragged it over to in the video), or the three quarter position (in which three-quarters of the new page are showing along with one quarter of the first).

### Problem to be solved

It is very frustrating to have to drag the paper back manually when I want to finish with all staves before moving to the next measure.

### Additional context

Though it would solve the problem; I don't like to use continuous view, especially if I am working on a final engraving of a score.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by MarcSabatella** [comment]
I agree there is room for refinement of how the repositioning works.  However, you certainly shouldn't need to resort to dragging in MsueScore - hardly ever.  Simplest way to return to the previous measure is simply press Left.  Or if you prefer using the mouse, use your scroll wheel or equivalent touch gesture to scroll back (with Shift if your device doesn't directly support horizontal scrolling as a native feature).

**Comment by bkunda** [comment]
I'd classify this more as a bug than a feature request. We need to stop the score from jumping around to a place that doesn't make sense. 

Another part of this is how the score jumps to a different position when inputting notes. It'd be nice if this wasn't necessary. 

https://github.com/user-attachments/assets/d95073c7-9204-4bfc-88a7-0811c6f8a807

**Comment by bkunda** [comment]
As starting points for some rules, I'd suggest: 

- For as long as the note input cursor remains in view, don't reposition the score when inputting notes
- If the note input cursor advances to a beat that's out of view, only reposition the score when the note is inputted. 

Then I think we can come up with some more granular behaviour based on the number of systems on a page. For example, if there's only one system (E..g full orchestral score), then showing 50% of each page could be nice. Just a light touch design could be helpful here @avvvvve.

**Comment by MarcSabatella** [comment]
In the past there were requests to always keep the current measure fully in view.  So, to not wait to reposition the view until cursor goes off screen, but do so as soon as you enter a measure that is not already fully in view.  This doesn’t always make sense, and the current behavior isn’t necessarily exactly that, but anyhow, it’s something to keep in mind.

**Comment by FrancRos31** [comment]
@Fourniture this issue has inevitably to do with #23901
